contrib/untar.c

```diff
@@ -36,6 +36,17 @@
 /* This is for mkdir(); this may need to be changed for some platforms. */
 #include <sys/stat.h>  /* For mkdir() */
 
+#ifdef _MSC_VER
+#define NUM_FORMAT "zu"
+#elif defined(__linux__) || defined(linux) || defined(__linux)
+#define NUM_FORMAT "d"
+#else
+#define NUM_FORMAT "lu"
+#endif /* _MSC_VER */
```

Comment:
Is there a special reason why not using %zu for all platforms?
And if not why %d on Linux and %lu elsewhere?

Reply:
`size_t` is different on different compilers and OSs

Reply:
And because of this the "z" length modifier was introduced in C99 to represent a size_t (on Linux supported since glibc 2.1)

Reply:
My next PR I was planning to contribute to libarchive is to backport it to C89.

Reply:
You're right that `size_t` is different in different places.  That means that this is going to get unwieldy if you really want to port to everything in the world.  ;-)

Instead, why not make `bytes_read` below be `unsigned` (since you know it will always be less than 512), and then you can hardcode `%u` in the printf strings.